subprojects/language-native/src/main/java/org/gradle/language/swift/tasks/SymbolHider.java

```diff
@@ -0,0 +1,158 @@
+ * Parse and hide given symbols in on object file based on COFF format documented
+ * here: https://docs.microsoft.com/en-us/windows/desktop/debug/pe-format
+ */
+public class SymbolHider {
+    DataReader data;
+    private byte[] bytes;
+
+    class DataReader {
+        private byte[] data;
+        private int position = 0;
+
+        public DataReader(byte[] data) {
+            this.data = data;
+        }
+
+        public int getPosition() {
+            return position;
+        }
+
+        public void moveTo(int position) {
+            this.position = position;
+        }
+
+        public int readByte() {
+            return Byte.toUnsignedInt(data[position++]);
+        }
+
+        public int readWord() {
+            return readByte() | readByte() << 8;
+        }
+
+        public int readDoubleWord() {
+            return readByte() | readByte() << 8 | readByte() << 16 | readByte() << 24;
+        }
+
+        public byte[] readBytes(int count) {
+            byte[] sub = Arrays.copyOfRange(data, position, position + count);
+            position += count;
+            return sub;
+        }
+    }
+
+    class COFFHeader {
+        public int machine;
+        public int numberOfSections;
+        public int timeDateStamp;
+        public int pointerToSymbolTable;
+        public int numberOfSymbols;
+        public int sizeOfOptionalHeader;
+        public int characteristics;
+
+        public COFFHeader() {
+            machine = data.readWord();
+            numberOfSections = data.readWord();
+            timeDateStamp = data.readDoubleWord();
+            pointerToSymbolTable = data.readDoubleWord();
+            numberOfSymbols = data.readDoubleWord();
+            sizeOfOptionalHeader = data.readWord();
+            characteristics = data.readWord();
+        }
+    }
+
+    class SymbolRecord {
+        private int storageClass;
+        private byte[] name;
+        private int value;
+        private int sectionNumber;
+        private int type;
+        private int numberOfAuxSymbols;
+
+        public SymbolRecord() {
+            name = data.readBytes(8);
+            value = data.readDoubleWord();
+            sectionNumber = data.readWord();
+            type = data.readWord();
+            storageClass = data.readByte();
+            numberOfAuxSymbols = data.readByte();
+        }
+
+        public String getName() {
+            // We only need to hide "main", so only support short named symbols here.
+
+            int nullCharIndex = 0;
+            for (nullCharIndex = 0; nullCharIndex < name.length; ++nullCharIndex) {
+                if (name[nullCharIndex] == 0) {
+                    break;
+                }
+            }
+            return new String(name, 0, nullCharIndex, StandardCharsets.UTF_8);
+        }
+    }
+
+    class SymbolTable {
+        private int numberOfSymbols;
+        private static final int IMAGE_SYM_CLASS_STATIC = 0x3;
+
+        public SymbolTable(int numberOfSymbols) {
```

Comment:
Same as above - make static and pass in the `DataReader`

Reply:
done